src/mame/drivers/diamondking.cpp

```diff
@@ -0,0 +1,176 @@
+      ||N| 8583N __________   |___________|             |_|  |N ||
+      | ____  |K6T0808C10 |                              __  |  ||
+      |74HC140|___________|                             | |  |  ||
+      | _____                                           | |  |  ||
+      ||CONN I2C                                        |_|  |__||
+      |    _____   ________________                   _____      |
+      |   |CONN|  |__CONN__________|              CONN RS232     |
+      |__________________________________________________________|
+
+****************************************************************************/
+
+#include "emu.h"
+
+#include "cpu/i8085/i8085.h"
+#include "machine/68340.h"
+#include "sound/ay8910.h"
+#include "sound/msm5205.h"
+#include "speaker.h"
+
+namespace
+{
+
+class diamondking_state : public driver_device
+{
+public:
+	diamondking_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_iocpu(*this, "iocpu")
+	{
+	}
+
+	void diamondking(machine_config &config);
+
+private:
+	required_device<cpu_device> m_maincpu;
+	required_device<m68340_cpu_device> m_iocpu;
+};
+
+static INPUT_PORTS_START(diamondking)
+	// On main board
+	PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+
+	// On I/O board
+	PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW2:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW2:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW2:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW2:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW2:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW2:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW2:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW2:8")
+
+	// On I/O board
+	PORT_START("DSW3")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW3:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW3:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW3:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW3:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW3:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW3:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW3:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW3:8")
+INPUT_PORTS_END
+
+void diamondking_state::diamondking(machine_config &config)
+{
+	I8085A(config, m_maincpu, 6.144_MHz_XTAL);
+
+	SPEAKER(config, "mono").front_center();
+
+	ay8910_device &ay8910(AY8910(config, "ay8910", 2'000'000)); // Frecuency unknown
+	ay8910.port_a_read_callback().set_ioport("DSW1");
+	ay8910.add_route(ALL_OUTPUTS, "mono", 1.0); // Guess
+
+	M68340(config, m_iocpu, 16'000'000); // Frecuency unknown
+
+	msm6585_device &msm6585(MSM6585(config, "msm6585", 0'640'000)); // Frecuency unknown
```

Comment:
You’ve made an octal constant - that's about 213kHz now, which is probably not what you meant.

Reply:
correct spelling is with a Q, frequency

Reply:
Fix spelling and OKI speed, thanks a lot.